lib/charms/operator_libs_linux/v2/snap.py

```diff
@@ -584,13 +584,16 @@ def ensure(
                     "Installing snap %s, revision %s, tracking %s", self._name, revision, channel
                 )
                 self._install(channel, cohort, revision)
-            else:
+                logger.info("The snap installation completed successfully")
+            elif (revision is None) or (revision != self._revision):
                 # The snap is installed, but we are changing it (e.g., switching channels).
                 logger.info(
                     "Refreshing snap %s, revision %s, tracking %s", self._name, revision, channel
                 )
                 self._refresh(channel=channel, cohort=cohort, revision=revision, devmode=devmode)
-            logger.info("The snap installation completed successfully")
+                logger.info("The snap refresh completed successfully")
+            else:
+                logger.info("Snap refresh was unnecessary %s", self._name)
```

Comment:
I think this log message won't be as readable as it could be, as the snap name is just tacked on the end. Can we set it off in some way, or perhaps reorder like so?

```suggestion
                logger.info("Refresh of snap %s was unnecessary", self._name)
```